### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by Ouail-01**
> ### In this pull request, I will Setup a Redux store & Create a slice to save my state and reducers

# Learning objectives :
- Understand the concept of state management.
- Use store, actions, and reducers in React.

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import categoriesReduce from './categories/categoriesSlice';
+import booksReduce from './books/booksSlice';
+
+export default combineReducers({
+  categoriesReduce,
+  booksReduce,
```
Comment (OLIPLICHE) [inline src/redux/configureStore.js:7]: - [ ] I am reading from the requirements of the project under the section `Configure the Redux Store (/src/redux/configureStore.js)`:
- It says to import the necessary methods from Redux Toolkit. I would you to import your  method( Example `configureStore`) from @reduxjs/toolkit
Example: `import { configureStore } from "@reduxjs/toolkit"`
- Then you can use `configureStore` to create a reducer: `books` and `categories` should be added to `configureStore` method. like what you did with `combineReducers` you created.



Example:
 ```jsx
const store = configureStore({
  reducer: { books: booksReducer, categories: categoriesReducer },
})

export default store
```

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const ADDBOOK = 'bookstore/books/ADDBOOK';
+const REMOVEBOOK = 'bookstore/books/REMOVEBOOK';
+
+const booksReduce = (state = {}, action) => {
```
Comment (OLIPLICHE) [inline src/redux/books/booksSlice.js:4]: - [ ] In reading from the requirement, it says Set the initial state to be an empty array of books. Therefore I would advise you to set state = [ ] for example, according to your code.

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Ouail-01, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linter is green (the last commit has a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏

## Required Changes ♻️


- [ ] I would advise you to make sure that adding and removing actions need to be implemented.


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @Ouail-01    :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**